sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -402,7 +405,28 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
       }
       Aggregate(groupByAttrs, patchedAggExpressions, firstAggregate)
     } else {
-      a
+      // We may have one distinct group only because we grouped using ExpressionSet.
+      // To prevent SparkStrategies from complaining during sanity check, we need to check whether
+      // the original list of aggregate expressions had multiple distinct groups and, if so,
+      // patch that list so we have only one distinct group.
```

Comment:
I don't quite understand this. Shall we use `ExpressionSet` to fix issues in `SparkStrategies` as well?

Reply:
>Shall we use ExpressionSet to fix issues in SparkStrategies as well?

Looking...